lib/collection/src/shards/local_shard/shard_ops.rs

```diff
@@ -247,7 +247,20 @@ impl ShardOperation for LocalShard {
             let update_sender = self.update_sender.load();
             let channel_permit = update_sender.reserve().await?;
 
-            let (operation_id, wal_lock) = self.wal.lock_and_write(&mut operation).await?;
+            // TODO: Refactor `lock_and_write`, so this is less terrible? :/
+            let (operation_id, wal_lock) = match self.wal.lock_and_write(&mut operation).await {
+                Ok(id_and_lock) => id_and_lock,
+
+                Err(crate::wal::WalError::Rejected) => {
+                    return Ok(UpdateResult {
+                        operation_id: None,
+                        status: UpdateStatus::Acknowledged,
+                        clock_tag: operation.clock_tag,
+                    });
```

Comment:
So, this is the "hack". I just use `operation_id: None` and `status: Acknowledged` and updated clock there. I think it's "good enough" to implement the retry code for now.

Reply:
Can't we introduce new `UpdateStatus`? It should be backward compatible, I think

Reply:
I think we can, just didn't want to bother with it _yet_.

Reply:
I've added this in <https://github.com/qdrant/qdrant/pull/3725>.